stream-chat-android-core/src/main/java/io/getstream/chat/android/core/utils/date/DateUtils.kt

```diff
@@ -47,3 +49,12 @@ public fun max(dateA: Date?, dateB: Date?): Date? = when (dateA after dateB) {
  */
 @InternalStreamChatApi
 public fun Date.inOffsetWith(other: Date, offset: Long): Boolean = (time + offset) >= other.time
+
+/**
+ * Returns difference between [this] date and the [otherTime] in [TimeDuration].
+ */
+@InternalStreamChatApi
+public fun Date.diff(otherTime: Long): TimeDuration {
+    val diff = abs(time - otherTime)
+    return TimeDuration.millis(diff)
```

Comment:
Do we need a custom `TimeDuration` class? I think this adds more code that we need to maintain and understand 🤔 . There is `java.time.Duration` which I think serves the same purpose, you can do this:

```
@InternalStreamChatApi
public fun Date.diff(otherTime: Long): Duration {
    val diff = abs(time - otherTime)
    return Duration.ofMillis(diff)
}
```

Reply:
Yeah, I didn't want to expose `kotlin.time.Duration` since we have Java users and `java.time.Duration` is available from API 26